File: Exercism Codes/Python/robot_name.py
import random


registeredName = set()


def generate_name():
    return random_name('A', 'Z', 2) + random_name('0', '9', 3)


def random_name(start, end, length):
    return ''.join(chr(random.randint(ord(start), ord(end))) for x in range(length))


class Robot:
    def __init__(self):
        self.name = None
        self.reset()
    
    def reset(self):
        while (name := generate_name()) in registeredName:
            continue

        registeredName.add(name)
        self.name = name

File: Exercism Codes/C#/RobotName.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Robot
{
    private static readonly HashSet<string> registered = new();
    private readonly Random random = new();

    public Robot() => Reset();

    public string Name { get; private set; }

    public void Reset()
    {
        string name;
        do
        {
            name = GenerateName();
        } while (registered.Contains(name));
        registered.Add(name);
        Name = name;
    }

    private string GenerateName() => RandomName('A', 'Z', 2) + RandomName('0', '9', 3);

    private string RandomName(char start, char end, int length) => string
        .Join("", Enumerable.Range(0, length)
        .Select(x => (char)random.Next(start, end + 1)));
}